src/main/kotlin/com/safehill/kclient/tasks/outbound/UploadOperationImpl.kt

```diff
@@ -265,10 +265,10 @@ class UploadOperationImpl(
         try {
             for (recipient in outboundQueueItem.recipients) {
                 val (_, sharablePayload) = encrypter.getSharablePayload(outboundQueueItem, user, recipient)
-                serverShare(outboundQueueItem, recipient, sharablePayload)
-                //todo use listeners to create a thread instead of upload operation creating it.
-                userInteractionController.setUpThread(outboundQueueItem.recipients, listOf())
+
+                val threadOutputDTO = userInteractionController.setUpThread(outboundQueueItem.recipients, listOf())
                     .also { threadStateRegistry.upsertThreadStates(listOf(it)) }
+                serverShare(outboundQueueItem, recipient, sharablePayload, threadOutputDTO.threadId)
```

Comment:
Could you remove this coupling on how we are storing thread in the android with the upload operation? Can we achieve it with listeners?

Also.. we should not upsert thread in registry directly.. This will remove already existing(if one exists) thread messages and asset identifiers with the newly created state.